### DISCUSSION TITLE
unable to pass parameter to Sql query in apache superset

### DISCUSSION THREAD

**Discussion by marituhone** [apache/superset#28136] [Q&A / Help]
@craig-rueda @dpgaspar @nytai  @rusackas can someone help?   I am using  superset  for the first time and i want to write my custom SQL that i want to use in my dashboard since there calculation is involved,that is not handled by slice, i want to handle it through SQL query,but after writing SQL in jinja i can't  parameterized i can't run it and get result since its parameterized and also i am unable to pass parameter

**Comment by rusackas** [comment]
https://preset.io/blog/intro-jinja-templating-apache-superset/
https://docs.preset.io/docs/parametrizing-queries-using-jinja
https://www.youtube.com/watch?v=CdZy3LivkhQ

Hopefully these help!

**Comment by marituhone** [comment]
@rusackas  thank you for replaying,but the thing is i do use these approaches  and  in the sql lab it works fine  it returned data based on the filter param but the problem is when  i try to change to it to chart after seeing the result  it gives nothing no data,no visualization and it only includes the columns i passed from the sql so since there is no data i can't even add filter.i can share what am trying to do ,
SELECT 
        seller_tin AS tin,
        SUM(CASE WHEN ev.buyer_tin = '{{ "'" + "','".join(filter_values('seller_tin')) + "'" }}' THEN ev.item_qty ELSE 0 END) AS purchased_quantity,
        SUM(CASE WHEN ev.seller_tin = '{{ "'" + "','".join(filter_values('seller_tin')) + "'" }}' THEN ev.item_qty ELSE 0 END) AS sold_quantity,
        SUM(CASE WHEN ev.buyer_tin = '{{ "'" + "','".join(filter_values('seller_tin')) + "'" }}' THEN ev.item_qty ELSE 0 END) - 
        SUM(CASE WHEN ev.seller_tin = '{{ "'" + "','".join(filter_values('seller_tin')) + "'" }}' THEN ev.item_qty ELSE 0 END) AS stock,
        ev.item_prd_desc 
    FROM
        eirp_view ev
    WHERE
        '{{ "'" + "','".join(filter_values('seller_tin')) + "'" }}' IN (ev.buyer_tin, ev.seller_tin)
    GROUP BY
        ev.item_prd_desc, seller_tin;        

but i cant pass parameter dynamically when i gave it static value in jinja template it  gives data but when i try to visualize it and link to filter  it doesn't work it says 0 rows  ? or do metrics support such kind of queries can i finish the process only using the slice  and dashboard?
